Observer/TruemoneyDataAssignObserver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace Omise\Payment\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Payment\Observer\AbstractDataAssignObserver;
+use Magento\Quote\Api\Data\PaymentInterface;
+
+class TruemoneyDataAssignObserver extends AbstractDataAssignObserver
+{
+    /**
+     * @var string
+     */
+    const PHONE_NUM   = 'truemoney_phone_num';
```

Comment:
Naming feels wrong since this isn't actually a phone number

Reply:
how would you name it?

Reply:
Maybe
```php
const PHONE_NUMBER = 'truemoney_phone_number';
```

Because `phone_number` has been used all over the code and `phone_num` is just only in this place.

Reply:
Maybe something like PHONE_NUMBER_FIELD? The name phone_num or phone_number suggests the constant should be a phone number, which it isn't

Reply:
changed to `num` => `number`

Reply:
no reason to add there `FIELD` suffix.
the current name is understandable and keeps a user's phone number.

Reply:
It doesn't hold the user's phone number. The constant holds the name of the field that holds the user's phone number

Reply:
it is hard not to agree.. .but we keep this naming convention in the entire plugin.

Reply:
My bad, it was started by me adding a constant to reflect with its field name just to make it looks obvious to see. But didn't think of it, that is not actually contain a value, just a field name. 
i.e.
```php
$paymentInfo[self::SOURCE] = [
    self::SOURCE_TYPE => 'alipay',
];
```

Think of this again, calling `Omise\Payment\Gateway\Request\AlipayBuilder::SOURCE_TYPE` and get `source_type`, look a bit off.
Please feel free to fix the entire convention if necessary.

Reply:
will definitely consider in the future.

Reply:
> it is hard not to agree.. .but we keep this naming convention in the entire plugin.

We should always endeavour to fix anything with code as we find the problems. Putting stuff off like this is a bad way to go about things. If we keep 'considering things' for the future, they will almost certainly not get done. I've seen this many, many times in the past. It gets to a point where either all the proposed changes are forgotten about, or the 'can kicking' has gotten to the point where the amount of 'technical debt' incurred is too intimidating and no-one wants to do the fixing any more as it is a big job.

Less work is required overall if we fix things as we find them. We should, as far as possible, keep our house tidy

Reply:
the reason is that I don't see it as any big thing. That is the reason.
Business cases are way more important for me then naming convention that is not maybe the best but we still kept it for many PRs unnoticed, and for me, it is very completely understandable.

Reply:
All of the subjects to refactor are for me future cases, don't see any of them as a threat for any complexity. yes, we have more javascript files, but a number of javascript files are a way more minor thing than allowing our customers to make money on ie. Convenience Store Payment.

That all refactorings in Magento 2 will be subjects of future stuff after delivering all business cases.

Reply:
This is precisely my point, and in no way am I just referring to this particular issue with naming convention. We don't do things as we don't consider them 'big things' - and before we know it, we have a huge pile of 'little' things that all really need attention. I have also pointed out in this PR the code has been added on top of other code that is in rather large need of refactoring, and this refactoring too was 'considered' for the future after previous PRs, but has not been done. This has the effect of increasing the amount of technical debt, and increasing the total amount of work that has to be done on the project overall

Reply:
By not doing the refactoring as we go, we are creating more work - and potentially delaying/complicating future business cases.

It isn't something we should do later as a chore, it should be an ongoing part of the development process

Reply:
well, that's are your feelings.
I don't see it as any threat of any complexity, and we have different opinions in naming, for instance, something you call: a `huge` pile of 'little' things, it is for me it is a `huge` exaggeration.

is there anything that holds you guys from approving this PR?

Reply:
I was talking in the general case about where this style of development usually leads. I've seen this happen too many times before to ignore it.

Presentations from Omise's CTO also encourage us to work in this manner

Reply:
ok, is there anything that holds you guys from approving this PR?